src/mapport.cpp

```diff
@@ -161,8 +161,11 @@ static bool ProcessUpnp()
     struct UPNPUrls urls;
     struct IGDdatas data;
     int r;
-
+#if MINIUPNPC_API_VERSION <= 17
```

Comment:
nit: I guess given we assert >= 17, this could be = 17, but I'm not sure that is less confusing.